src/components/SiderMenu/BaseMenu.tsx

```diff
@@ -14,7 +15,7 @@ const { SubMenu } = Menu
 //   icon: 'setting',
 //   icon: 'http://demo.com/icon.png',
 //   icon: <Icon type="setting" />,
-const getIcon = icon => {
+const getIcon = (icon: string) => {
```

Comment:
it seams that `icon` can be a component as well. Please update the type.

- [ ] update type for icon

Reply:
The usage here is incorrect. The base Icon class from antd doesnt accept type prop on it.

``return <Icon type={icon}> ``

Reply:
But look at the last `return` and the `if`.
Also why did you comment out the `else` part? From what I see the `type` is a correct attribute: https://ant.design/components/icon/
If you have a weird type error then you can add `// @ts-ignore` next to that line to ignore a typecheck.